src/makim/cli.py

```diff
@@ -1,199 +1,150 @@
-        )
+from typer import Argument, Option, Typer, echo
+from typing_extensions import Annotated
 
+from makim import Makim, __version__
 
 makim = Makim()
-
-
-def _get_args():
-    """
-    Define the arguments for the CLI.
-
-    note: when added new flags, update the list of flags to be
-          skipped at extract_makim_args function.
-    """
-    makim_file_default = str(Path(os.getcwd()) / '.makim.yaml')
-
-    parser = argparse.ArgumentParser(
-        prog='Makim',
-        description=(
-            'Makim is a tool that helps you to organize '
-            'and simplify your helper commands.'
-        ),
-        epilog=(
-            'If you have any problem, open an issue at: '
-            'https://github.com/osl-incubator/makim'
-        ),
-        add_help=False,
-        formatter_class=CustomHelpFormatter,
-    )
-    parser.add_argument(
+app = Typer(
+    name='Makim',
+    epilog=(
+        'If you have any problem, open an issue at: '
+        'https://github.com/osl-incubator/makim'
+    ),
+)
+makim_file_default = str(Path(os.getcwd()) / '.makim.yaml')
+
+
+@app.callback(invoke_without_command=True)
+def main(
+    # Common Options
+    help: bool = Option(
+        None,
         '--help',
         '-h',
-        action='store_true',
-        help='Show the help menu',
-    )
-
-    parser.add_argument(
+        is_flag=True,
+        help='Show the version and exit',
+    ),
+    version: bool = Option(
+        None,
         '--version',
-        action='store_true',
-        help='Show the version of the installed Makim tool.',
-    )
-
-    parser.add_argument(
+        '-v',
+        is_flag=True,
+        help='Show the version and exit',
+    ),
+    verbose: bool = Option(
+        None,
         '--verbose',
-        action='store_true',
+        is_flag=True,
         help='Show the commands to be executed.',
-    )
-
-    parser.add_argument(
+    ),
+    dry_run: bool = Option(
+        None,
         '--dry-run',
-        action='store_true',
+        is_flag=True,
         help="Show the commands but don't execute them.",
-    )
+    ),
+    # Makim-specific Options
+    makim_file: Annotated[
+        str,
+        Option(
+            '--makim-file',
+            help='Specify a custom location for the makim file.',
+            is_flag=True,
+        ),
+    ] = makim_file_default,
+    target: Annotated[
+        str,
+        Argument(
+            ...,
+            help='Specify the target command to be performed.',
+        ),
+    ] = '',
+):
+    """
+    Makim is a tool.
+
+    that helps you to organize and simplify your helper commands.
+    """
+    if version:
+        return show_version()
+
+    if help or not target:
+        return create_help_text(makim_file)
+
+    args = {
+        'dry_run': dry_run,
+        'help': help,
+        'makim_file': makim_file,
+        'target': target,
+        'verbose': verbose,
+        'version': version,
+    }
+
+    makim.load(makim_file)
```

Comment:
not sure if that is the write way to work with makim and typer together

in this main function you probably should allow makim-file, and version.

for the other commands you will need to implement (for each one) verbose, dry-run and makim-file

the target you can set manually for each one, because each command will trigger just one function

does it make sense?

Reply:
btw, let me know if this one is too complex for you ... and I can work on this one.
this one is really tricky because it is not using the default workflow for typer

Reply:
Let me try once more time, I will report to you.